common/persistence/serialization/task_serializer_test.go

```diff
@@ -0,0 +1,332 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package serialization
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/persistence"
+)
+
+func TestTaskSerializerThriftRW(t *testing.T) {
+	parser, err := NewParser(common.EncodingTypeThriftRW, common.EncodingTypeThriftRW)
+	require.NoError(t, err)
+	taskSerializer := NewTaskSerializer(parser)
+
+	workflowIdentifier := persistence.WorkflowIdentifier{
+		DomainID:   "8be8a310-7d20-483e-a5d2-48659dc47602",
+		WorkflowID: "test-workflow",
+		RunID:      "4be8a310-7d20-483e-a5d2-48659dc47609",
+	}
+
+	testCases := []struct {
+		category persistence.HistoryTaskCategory
+		task     persistence.Task
```

Comment:
I would suggest generating task data so that new fields added will be caught by the test if there's a mapping problem. see `common/testing/testdatagen` if it's useful.